java-checks/src/test/java/org/sonar/java/checks/targets/UnusedPrivateMethod.java

```diff
@@ -51,26 +51,41 @@ private void readObjectNoData() throws java.io.ObjectStreamException {
   }
 
   @SuppressWarnings("unused")
-  private int unusedPrivateMethod() {
+  private int unusedPrivateMethod() { // Noncompliant {{Private method 'unusedPrivateMethod' is never used.}}
     return 1;
   }
-  private int unusedPrivateMethod(int a, String s) {
+
+  private int unusedPrivateMethod(int a, String s) { // Noncompliant {{Private method 'unusedPrivateMethod' is never used.}}
     return 1;
   }
 
+  private void variadic(String first, Object... objects) {
+  }
+
+  private void variadic(String... strings) { // Noncompliant {{Private method 'variadic' is never used.}}
+    // this is a false positive
+  }
+
+  public void usage() {
+    variadic("", new Object(), new Object());
+    variadic("", "", ""); // should resolve to 'String...' and not 'String, Object...'
```

Comment:
This should lead to a creation of the corresponding jira ticket to fix semantic analysis.